sdk/monitor/azure-monitor-opentelemetry-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/implementation/pipeline/TelemetryPipeline.java

```diff
@@ -137,10 +152,20 @@ private void onResponseBody(
             return;
         }
 
-        listener.onResponse(request, new TelemetryPipelineResponse(responseCode, responseBody));
+        TelemetryPipelineResponse telemetryPipelineResponse = new TelemetryPipelineResponse(responseCode, responseBody);
+        listener.onResponse(request, telemetryPipelineResponse);
         if (responseCode == 200) {
             result.succeed();
         } else {
+            if (responseCode == 400
+                && statsbeatModule != null
+                && telemetryPipelineResponse.invalidInstrumentationKey()) {
+                LOGGER.warning("400 status code is returned for an invalid customer's instrumentation key. Shutting down Statsbeat and its listeners.");
+                statsbeatModule.shutdown();
+                if (listener instanceof LocalStorageTelemetryPipelineListener) {
+                    listener.shutdown();
```

Comment:
@trask do we need to shutdown profiler when ikey is invalid? maybe a question for john oliver?